app/models/metadatum.rb

```diff
@@ -0,0 +1,114 @@
+class Metadatum < ApplicationRecord
+  # ActiveRecord related
+  belongs_to :sample
+  # When using an ActiveRecord enum, the type returned from reading records is String.
+  enum data_type: [:string, :number]
+  validates :text_validated_value, length: { maximum: 250 }
+  validates :number_raw_value, :number_validated_value, numericality: true, allow_nil: true
+  validate :set_validated_values
+
+  STRING_TYPE = 0
+  NUMBER_TYPE = 1
```

Comment:
you did not find a way to reference it for other class variables?
if there is no alternative do at least: `
 STRING_TYPE = 0
 NUMBER_TYPE = 1
enum data_type: {:string => STRING_TYPE, :number => NUMBER_TYPE}
`
So that you do not have to keep the two lists in sync.

Reply:
yeah I couldn't find a way to use `enum` directly apparently. Changed to above